pkg/cmd/ssh-key/add/http.go

```diff
@@ -3,21 +3,50 @@ package add
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
 	"net/http"
+	"strings"
 
 	"github.com/cli/cli/v2/api"
 	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/pkg/cmd/ssh-key/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
 )
 
-func SSHKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader, title string) error {
+func SSHKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader, title string, printSuccessMsgs bool, ios *iostreams.IOStreams) error {
```

Comment:
Instead of `printSuccessMsgs` I might prefer just checking for `ios` being `nil`. What do you think?

Reply:
That makes sense, I'll make that change.

Reply:
Actually, an error is printed a few lines down that assumes `ios` won't be `nil`. Should I rework this a bit so the error is printed a level up, or just leave this parameter?

Reply:
ah right, I missed the error printing. I think the solution here might be to just let this function return an error for that case and then let its caller decide whether or not to print something to std error. The caller would also be responsible for printing success messages. I'd rather see potential duplication of success message printing than extra i/o stuff passed into a helper like this.

Reply:
Works for me, I just pushed that change. Thanks!